### PR TITLE
[Bug] Fix Pokemon not gaining HP when evolving

### LINKED ISSUE #2352: [BUG] HP not displaying correctly after evolution
This really bizarre bug happened while I wasn't paying attention.
So I was fighting Surge, and my Toxtricity evolved. But when I had to send it out, until he hit me, its HP says 71/71, but its bar displays a bit of damage. It leveled up immediately after and everything was resolved (HP at proper max of 95)
PokerogueWhyAreYouSoBroken

### PR DESCRIPTION
**Pull request by DayKev**
## What are the changes?
When a Pokémon evolves its current HP will increase by the same amount that its max HP does.

## Why am I doing these changes?
Fixes #2352

## What did change?
When creating a new `Pokemon` object, `calculateStats()` is not called if already existing stats are passed through the constructor.

### Screenshots/Videos
Before fix:

https://github.com/user-attachments/assets/2081477d-151b-4d10-8728-81ff7b8c7453

After fix:

https://github.com/user-attachments/assets/46c9462c-9442-406e-96bf-14a07d67c848

## How to test the changes?
Evolve a Pokémon in the same conditions pre- and post-patch.

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- ~[ ] Are the changes visual?~
  - ~[ ] Have I provided screenshots/videos of the changes?~

### CODE REVIEW

File: src/test/utils/mocks/mockVideoGameObject.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { vi } from "vitest";
+import { MockGameObject } from "./mockGameObject";
+
+export class MockVideoGameObject implements MockGameObject {
```
Comment (torranx) [inline src/test/utils/mocks/mockVideoGameObject.ts:4]: I know this is a mock class but can we at least add some docs/description on it?

**Comment by flx-sta** [comment]
@DayKev there is 1 more thing to be tested: when it evolved without having full hP

**Comment by flx-sta** [comment]
Added a test for that too
https://github.com/DayKev/pokerogue/pull/5

**Comment by DayKev** [comment]
Ugh Github why are you like this?

**Review by flx-sta — CHANGES_REQUESTED** [review]
You imported every phases file with `.js` extension…

**Review by flx-sta — DISMISSED** [review]

File: src/test/utils/mocks/mockVideoGameObject.ts (hunk 1 above)
Comment (DayKev) [reply src/test/utils/mocks/mockVideoGameObject.ts:4]: I'm unsure what docs there even are to put here. @flx-sta

**Review by innerthunder — DISMISSED** [review]

**Review by frutescens — APPROVED** [review]

File: src/test/evolution.test.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
 import { pokemonEvolutions, SpeciesFormEvolution, SpeciesWildEvolutionDelay } from "#app/data/pokemon-evolutions";
 import { Abilities } from "#app/enums/abilities";
+import { Moves } from "#app/enums/moves.js";
```
Comment (flx-sta) [inline src/test/evolution.test.ts:3]:
```suggestion
import { Moves } from "#app/enums/moves";
```

File: src/test/utils/mocks/mockVideoGameObject.ts (hunk 1 above)
Comment (flx-sta) [reply src/test/utils/mocks/mockVideoGameObject.ts:4]: Something that links to the original interface.  Nothing fancy

File: src/test/evolution.test.ts (hunk 2 above)
Comment (DayKev) [reply src/test/evolution.test.ts]: There's gotta be a way to make auto-import not add `.js` to the imports...

File: src/test/utils/mocks/mockVideoGameObject.ts (hunk 1 above)
Comment (DayKev) [reply src/test/utils/mocks/mockVideoGameObject.ts:4]: Well that's the thing, I'm not sure what exactly the original interface is; this is from your PR.

**Review by innerthunder — APPROVED** [review]

**Review by Tempo-anon — APPROVED** [review]

**Review by frutescens — APPROVED** [review]

**Pull request merged**